core/templates/domain/skill/skill-creation-backend-api.service.ts

```diff
@@ -29,46 +30,75 @@ export interface ISkillCreationBackend {
   description: string,
   'explanation_dict': string,
   'linked_topic_ids': string[],
-  rubrics: IRubricBackend
+  rubrics: IRubricBackend,
+  filenames: string[]
+}
+
+export interface IImageData {
+  filename: string,
+  imageBlob: Blob
 }
 
 @Injectable({
   providedIn: 'root'
 })
 export class SkillCreationBackendApiService {
-  constructor(private http: HttpClient) {}
+  constructor(private csrfTokenService: CsrfTokenService) {}
 
   _createSkill(
       successCallback: (value?: Object | PromiseLike<Object>) => void,
       errorCallback:(reason?: any) => void,
       description: string, rubrics: IRubricBackend, explanation: string,
-      linkedTopicIds: string[]): void {
+      linkedTopicIds: string[], imagesData: IImageData[]): void {
+    let filenames = imagesData.map(obj => obj.filename);
     let postData:ISkillCreationBackend = {
       description: description,
       linked_topic_ids: linkedTopicIds,
       explanation_dict: explanation,
-      rubrics: rubrics
+      rubrics: rubrics,
+      filenames: filenames
     };
-    this.http.post(
-      '/skill_editor_handler/create_new', postData).toPromise()
-      .then((response: { skillId: string }) => {
+    var form = new FormData();
```

Comment:
Should we define (line 61 to 82) in a service so that it can be reusable and we can avoid duplicate codes?

Reply:
Yeah, will do that. Something like 'image-uploader-backend-api-service' where all image upload related requests are sent?

Reply:
Is this still required? I am adding images in the request keyed by filename, future implementations need not use that method.

Reply:
Switched this back to use http itself since testing these ajax calls in an Angular 8 service had a lot of issues and the recommended way found online was to use another package called 'jasmine-ajax'.
So, figured out a way to use the existing http method itself and modified the interceptor accordingly (in fact, we can change all the ajax calls in the codebase where we send assets to the backend to use the http method itself).